baybe/utils/basic.py

```diff
@@ -284,3 +309,14 @@ def wraps(*args, **kwargs):
         return result
 
     return wraps
+
+
+def set_default_torch_dtype():
+    """Lazily set the torch default dtype."""
+    # TODO: This is only a temporary workaround. A proper solution requires
```

Comment:
@AVHopp, @Scienfitz: We need to properly fix this at some point, but yesterday I gave up after two hours of trying. I think for now, the priority is on fixing the bugs. Nonetheless, any thoughts from your end?

Reply:
Agree that fixing the bug is definitely top priority. Still, why can't this exact code not be inserted in `to_gpytorch`?

Reply:
It currently *is* set in `to_gpytorch`, and that works as a temporary solution, but it doesn't really fix the problem, because it requires us to always manually set the dtype (and first of all not forget to do so) whenever we load torch **somewhere**. As soon as you do an `import torch` in any module and you forget to immediately set the dtype afterward, your created tensors will be `float32`. So a proper solution is to change the import itself and make it automatically set the dtype to whatever we have defined via our environment variable. This is definitely possible, I just couldn't get it to work yet

Reply:
Ok, so this is a general issue which is just the most prominent resp. was just found here? In that case I agree that this should be fixed.

Reply:
dump question, is it possible to import torch *via* another module? Eg do
```python
# in utils.torch
import torch
torch.set_dtype(diesdas)
```

and use everywhere else `from baybe.utils.torch import torch` ?

Reply:
yes, is possible, but doesn't solve the core problem, i.e. now you to make sure that all your torch imports go via that module. So in both cases, you need to "remember" how to import torch...

Reply:
ok (only) now the corresponding entry in the PR description makes sense to me. 

Why do you consider the current workaround complete though? you only call the function in one place, what about all other places where torch is imported? If the new dtype utility is used only in 1 place, why don't you simply put the content where the utility is called and forget about the utility?

Reply:
That was pretty much for decoupling only, to have a single place where the workaround is implemented. I then wanted to add it wherever necessary to make the tests pass, but that seemed to be the only necessary place (at least according to what our tests cover). So I'm not sure what is the best strategy now. It's definitely not needed everywhere. If needed elsewhere at all, then only in places where tensors are created. So what is your suggestion, do you want to have it added other places as well?

Reply:
if just needed in one place the utility is pointless, but for consistency/safety it prob should be everywhere
if it should be everywhere I'd slightly prefer a custom import from utils instead of the function, but up to you. It needs to be mentioned in dev guidelines in both cases (in genreal I think we didnt even mention lazy loading there) 

which test is actually failing if you dont use this?
Wouldnt numpy require the same thing?

Reply:
No need for numpy currently. Also, for torch we usually don't run into problems because we explicitly set the dtype everywhere (which we potentially could even remove when we get the importlib stuff running, which would give us a safer alternative). The reason why we need it in this very special situation is because the tensor creation is not done by us but by gpytorch. For instance, if you create a `LogNormalPrior(0.1, 0.2).to_gpytorch()`, the passed `float64` values will turn into `torch.float32` internally.

I actually can't remember which test failed (and can't reproduce the failing test right now, perhaps it was a random one generated by hypothesis), but the problem definitely still exists.

Then let me just add the workaround locally to the prior code. Having it part of utils is not so nice because it then requires to isolated TODO notes, one in the prior code explaining why we import from utils instead of importing torch directly, and one in utils explaining the importlib solution